core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -67,6 +68,8 @@ private[ui] class StagePage(parent: StagesTab) extends WebUIPage("stage") {
   // if we find that it's okay.
   private val MAX_TIMELINE_TASKS = parent.conf.getInt("spark.ui.timeline.tasks.maximum", 1000)
 
+  private val displayPeakExecutionMemory =
+    parent.conf.getOption("spark.sql.unsafe.enabled").exists(_.toBoolean)
```

Comment:
The issue is that we don't track all SQL operators (e.g. normal non-unsafe in-memory Sort). It would be misleading if Sort ends up having lower "peak execution memory" than UnsafeExternalSort. Let's move this discussion to the main thread.